### PR TITLE
Feat: Golang Repo Scaffolding: implement other interfaces (uninstall/reinstall)

### PR DESCRIPTION
**Pull request by lfbdev**
# Summary

Add other features (uninstall / reinstall) for "Golang Repo Scaffolding"

uninstall: remove the repo

reinstall: remove the repo and recreate the repo

## Key Points

- [x] Documentation (new or existing) is updated.

### Related Issues

#191

### CODE REVIEW

File: docs/github-repo-scaffolding-golang_plugin.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This plugin installs github-repo-scaffolding-golang.
 
+tips:
```
Comment (daniel-hutao) [inline docs/github-repo-scaffolding-golang_plugin.md:5]: - tip1
- tip2
- ...

File: pkg/util/github/repo.go [hunk 2]
```diff
@@ -17,3 +21,34 @@ func (c *Client) CreateRepo() error {
...
+		c.Option.Repo)
+
+	if err != nil {
+		return err
+	}
+	log.Successf("GitHub repo %s removed.", c.Repo)
+	return nil
+}
+
+func (c *Client) CheckRepo() error {
```
Comment (daniel-hutao) [inline pkg/util/github/repo.go:38]: Maybe `IsRepoExists()` is better

**Review by IronCore864 — CHANGES_REQUESTED** [review]

File: internal/pkg/plugin/reposcaffolding/github/golang/ishealthy.go [hunk 3]
```diff
@@ -1,7 +1,44 @@
...
+	}
+
+	if errs := validate(&param); len(errs) != 0 {
+		for _, e := range errs {
+			log.Errorf("Param error: %s", e)
+		}
+		return false, fmt.Errorf("params are illegal")
+	}
+
+	return check(&param)
```
Comment (IronCore864) [inline internal/pkg/plugin/reposcaffolding/github/golang/ishealthy.go:26]: IsHealthy only checks the validity of the parameters, which doesn't seem enough. However, this is related to the state, and whether we shall use DTM as the single source of truth or not. It's a big decision; let's move this discussion offline.

File: docs/github-repo-scaffolding-golang_plugin.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lfbdev) [reply docs/github-repo-scaffolding-golang_plugin.md]: image

File: pkg/util/github/repo.go [hunk 4]
```diff
@@ -17,3 +21,34 @@ func (c *Client) CreateRepo() error {
 	}
 	return nil
 }
+
+func (c *Client) DeleteRepo() error {
+	_, err := c.Client.Repositories.Delete(
```
Comment (daniel-hutao) [inline pkg/util/github/repo.go:26]: just `_, err := c.Client.Repositories.Delete(c.Context, c.Owner, c.Repo)` is fine. same for below

File: pkg/util/github/repo.go [hunk 5]
```diff
@@ -17,3 +21,34 @@ func (c *Client) CreateRepo() error {
...
+
+	if err != nil {
+		return err
+	}
+	log.Successf("GitHub repo %s removed.", c.Repo)
+	return nil
+}
+
+func (c *Client) IsRepoExists() error {
+	rps, rsp, err := c.Client.Repositories.Get(
```
Comment (daniel-hutao) [inline pkg/util/github/repo.go:39]: maybe `resp` is better

**Review by daniel-hutao — APPROVED** [review]

**Review by IronCore864 — APPROVED** [review]

**Pull request merged**